File: lib/src/network/CancellationHandler.java
package com.alibaba.sdk.android.oss.network;

import okhttp3.Call;

/**
 * Created by zhouzhuo on 11/23/15.
 */
 class CancellationHandler {

      bool isCancelled;

      Call call;

     void cancel() {
        if (call != null) {
            call.cancel();
        }
        isCancelled = true;
    }

     bool isCancelled() {
        return isCancelled;
    }

     void setCall(Call call) {
        this.call = call;
    }
}

File: lib/src/network/cancellation_handler.dart
import 'package:dio/dio.dart';

class CancellationHandler {
  bool _isCancelled = false;

  CancelToken? call;

  void cancel() {
    call?.cancel();
    _isCancelled = true;
  }

  bool get isCancelled {
    return _isCancelled;
  }
}
